### PR TITLE
feat: add function history_query/1

### PR DESCRIPTION
**Pull request by brunohkbx**
As discussed in #64, history_query/1 offers much more flexibility than the old history/2

### CODE REVIEW

File: lib/repo/repo.ex [hunk 1]
```diff
@@ -179,13 +179,19 @@ defmodule ExAudit.Repo do
       end
 
       # ExAudit.Repo behaviour
+
+      @deprecated "Use history_query/1 instead"
       def history(struct, opts \\ []) do
```
Comment (brunohkbx) [inline lib/repo/repo.ex]: @narrowtux should we keep or deprecate this one?

File: lib/repo/queryable.ex [hunk 2]
```diff
@@ -74,6 +74,17 @@ defmodule ExAudit.Queryable do
     end
   end
 
+  def history_query(%{id: id, __struct__: struct}) do
+    import Ecto.Query
```
Comment (rauann) [inline lib/repo/queryable.ex]: @brunohkbx what do you think about add this `import` at the module level and only import the `from/1` function?

File: lib/repo/repo.ex (hunk 1 above)
Comment (narrowtux) [reply lib/repo/repo.ex]: Keep. The history function will also render complete structs from the diffs, but it needs the entire history for that.

**Comment by brunohkbx** [comment]
@narrowtux please check again when you have a chance :smiley:

**Pull request merged**